src/main/java/euteam/cleanmachine/controller/MachineController.java

```diff
@@ -1,29 +1,193 @@
+        //Prepare result into object
+        HashMap<String,String> result = new HashMap<>();
+        result.put("status",stateName);
+
+        return ResponseEntity.ok(result);
+    }
+
+    /**
+     * A Customer or employe logs in on the machine
+     * @param authItemToken (USER token that can be found on card)
+     * @return JSON
+     */
+    @RequestMapping(path="/machine/login", method = RequestMethod.POST)
+    public ResponseEntity<?> authenticateUserOnMachine(@RequestParam("authItemToken") Long authItemToken){
+        Machine machine = getMachineByAuthentication();
+
+            User u = userService.getUserByAuthId(authItemToken);
+        if(u== null){
+              return   ResponseEntity.status(400).body("No user found linked to this Authentication item");
+        }
+        try {
+            machine.authenticateOnMachine(u.getId());
+        }catch(StateTransitionException e){
+            return ResponseEntity.status(400).body("Cannot change to authentication state at the moment");
+        }
+        machineService.update(machine);
+
+        List<ProgramDto> programs = new ArrayList<>();
+        for (Program program: machine.getPrograms()) {
+            programs.add(DtoFactory.programToDto(program));
+        }
+        HashMap<String, Object> result = new HashMap<>();
+        result.put("Programs",programs );
+        result.put("Account Balance", accountService.getAccountByUser(u).getBalance());
+        return ResponseEntity.ok(result);
+    }
+
+    /**
+     * gives all programs and user balance, used to get to machine state after reboot;
+     * @return
+     */
+    @RequestMapping(path="/machine/getLoggedInInfo", method = RequestMethod.GET)
+    public ResponseEntity<?> getAuthenticateInfo(){
+        Machine machine = getMachineByAuthentication();
+        User u = userService.getUserByID(machine.getLoggedInUserId());
+        List<ProgramDto> programs = new ArrayList<>();
+        for (Program program: machine.getPrograms()) {
+            programs.add(DtoFactory.programToDto(program));
+        }
+        HashMap<String, Object> result = new HashMap<>();
+        result.put("Programs",programs );
+        result.put("Account Balance", accountService.getAccountByUser(u).getBalance());
+        return ResponseEntity.ok(result);
+    }
+
+    /**
+     * deauthenticates user from a machine
+     * @return
+     */
+    @RequestMapping(path="/machine/logout", method = RequestMethod.POST)
+    public void userLogout(){
+        Machine machine = getMachineByAuthentication();
+        try {
+            machine.idle();
+            machineService.update(machine);
+        }
+        catch(StateTransitionException exception){
+            //TODO return failed code
+        }
+
+    }
+
+    @RequestMapping(path="/machine/startProgram", method = RequestMethod.POST)
+    public ResponseEntity<?> startProgram(@RequestParam("programId") long programId){
+        Machine machine = getMachineByAuthentication();
```

Comment:
Use a service for the business logic.

Reply:
okay, I was thinking the same but wasn't sure

Reply:
👍